crates/matrix-sdk-ui/src/timeline/event_item/content/mod.rs

```diff
@@ -117,12 +117,10 @@ pub enum TimelineItemContent {
 }
 
 impl TimelineItemContent {
-    /// If the supplied event is suitable to be used as a `latest_event` in a
-    /// message preview, extract its contents and wrap it as a
-    /// `TimelineItemContent`.
-    pub(crate) fn from_latest_event_content(
-        event: AnySyncTimelineEvent,
-    ) -> Option<TimelineItemContent> {
+    /// If the supplied event is suitable to be used as a message preview
+    /// like reply previews or latest event items in the room list,
+    /// extract its contents and wrap it as a `TimelineItemContent`.
+    pub(crate) fn from_any_sync_event(event: AnySyncTimelineEvent) -> Option<TimelineItemContent> {
         match is_suitable_for_latest_event(&event) {
```

Comment:
Async in our DMs we agreed that it was possible to reuse the `from_latest_event` function to synthethise the event, in fact you even said it was okay to do the renaming. I even asked if in case it was not the correct way to address this function here: https://github.com/matrix-org/matrix-rust-sdk/pull/3439#discussion_r1622395128 what was the correct way to solve this issue, since I myself was unsure if that was okay,
Your suggestion now makes sense, but would have been nice to have this been pointed way before without the need of saying that I have ignored feedback, since I actually asked for help, and flagged my concern about reusing this function